superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/ColumnSelect.tsx

```diff
@@ -124,7 +131,8 @@ export function ColumnSelect({
 
   return (
     <Select
-      value={value}
+      mode={mode}
+      value={mode === 'multiple' ? value || [] : value}
```

Comment:
When testing it felt like this could make sense for the groupby filter at least:
```js
allowClear={mode === 'multiple'}
```